testbed/tests/app/test_screen.py

```diff
@@ -0,0 +1,60 @@
+import os
+from importlib import import_module
+
+import pytest
+
+from toga.platform import current_platform
+
+
+@pytest.fixture
+def screen_probe_list(app):
+    module = import_module("tests_backend.screen")
+    if current_platform == "android":
+        return [getattr(module, "ScreenProbe")(app, screen) for screen in app.screens]
+    else:
+        return [getattr(module, "ScreenProbe")(screen) for screen in app.screens]
+
+
+async def test_name(screen_probe_list):
+    """The name of the screens can be retrieved"""
+    for screen_probe in screen_probe_list:
+        assert isinstance(screen_probe.screen.name, str)
+        screen_probe.assert_name()
```

Comment:
As noted previously - the *assertion* shouldn't be in the probe unless there is something platform specific about the assertion. In this case, the assertion is "name == expected value". That's not platform specific; the only thing platform specific is how you *get* the name - and, as I noted in my previous review, that doesn't even need a probe in this case, because the public API will provide the information you need.

Reply:
Removed it.